sway-fmt-v2/src/utils/ty.rs

```diff
@@ -0,0 +1,71 @@
+use crate::fmt::{Format, FormattedCode, Formatter, FormatterError};
+use sway_parse::{
+    brackets::{Parens, SquareBrackets},
+    expr::Expr,
+    keywords::{StrToken, UnderscoreToken},
+    path::PathType,
+    ty::{Ty, TyArrayDescriptor, TyTupleDescriptor},
+};
+use sway_types::Spanned;
+impl Format for Ty {
+    fn format(
+        &self,
+        formatted_code: &mut FormattedCode,
+        _formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        match self {
+            Self::Array(arr_descriptor) => format_array(formatted_code, arr_descriptor.clone()),
+            Self::Infer { underscore_token } => format_infer(formatted_code, underscore_token),
+            Self::Path(path_ty) => format_path(formatted_code, path_ty),
+            Self::Str { str_token, length } => {
+                format_str(formatted_code, str_token.clone(), length.clone())
+            }
+            Self::Tuple(tup_descriptor) => format_tuple(formatted_code, tup_descriptor.clone()),
+        }
+    }
+}
+
```

Comment:
`Path`, `Tuple` and `Array` can all implement `Format` I think

Reply:
But I think it could be okay either way, I'll defer to your preference

Reply:
If you can, I think it would be cleaner

Reply:
For sure! Don't want to block you, just a suggestion to keep in mind that certain parts may benefit from using the trait instead of just duplicating the function for each one